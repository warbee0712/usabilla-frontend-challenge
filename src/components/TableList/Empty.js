import React from 'react'
import Box from '../../elements/Box'
import Text from '../../elements/Text'

const Empty = () => (
  <Box p={[75, 150]}>
    <Text fontSize="l" lineHeight="l" textAlign="center">
      Oops, we couldn&apos;t find anything with these filters. Perhaps you can
      try loosening the search parameters a little bit.
    </Text>
  </Box>
)

export default Empty
