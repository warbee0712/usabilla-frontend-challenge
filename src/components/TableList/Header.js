import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
// Elements
import Grid from '../../elements/Grid'
import Box from '../../elements/Box'
import Text from '../../elements/Text'

export const HeaderText = styled(Text).attrs({
  fontWeight: 500,
})`
  text-transform: lowercase;
  font-variant: small-caps;
`

const Header = ({ fields }) => {
  return (
    <Grid
      display={['none', 'grid']}
      gridAutoColumns={['1fr 4fr 2fr 2fr 2fr']}
      gridTemplateAreas='"rating comment browser device platform"'
      gridColumnGap="m"
      px="l"
      py="m"
    >
      {fields.map(field => (
        <Box key={field} gridArea={field}>
          <HeaderText>{field}</HeaderText>
        </Box>
      ))}
    </Grid>
  )
}

Header.propTypes = {
  fields: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
}

export default Header
