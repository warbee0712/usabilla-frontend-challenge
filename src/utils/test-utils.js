import React from 'react'
import { render } from 'react-testing-library'
import { ThemeProvider } from 'styled-components'
import theme from '../theme'

const customRender = (node, options) => {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>, options)
}

// re-export everything
export * from 'react-testing-library'

// override render method
export { customRender as render }
