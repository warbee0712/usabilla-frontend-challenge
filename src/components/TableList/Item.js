import React, { memo } from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
// Elements
import Grid from '../../elements/Grid'
import Flex from '../../elements/Flex'
import Box from '../../elements/Box'
import Text from '../../elements/Text'
// Components
import { HeaderText } from './Header'
import Badge from '../Badge'

const StyledGrid = styled(Grid)`
  border-bottom: 1px solid ${props => props.theme.colors.light4};
`

const SmallText = styled(Text).attrs({
  as: 'span',
  fontSize: 'xs',
  lineHeight: 'xs',
})``

const Item = memo(({ data }) => (
  <StyledGrid
    gridAutoColumns={['1fr', '1fr 4fr 2fr 2fr 2fr']}
    gridTemplateAreas={[
      '"rating" "comment" "browser" "device" "platform"',
      '"rating comment browser device platform"',
    ]}
    gridColumnGap="m"
    p="l"
    data-testid="item"
  >
    <Flex pb={['s', 0]} alignItems="baseline" gridArea="rating">
      <Box display={['inline-flex', 'none']} flex="0 100px">
        <HeaderText>rating</HeaderText>
      </Box>
      <Badge active data-testid="item-rating">
        {data.rating}
      </Badge>
    </Flex>
    <Flex pb={['s', 0]} alignItems="baseline" gridArea="comment">
      <Box display={['inline-flex', 'none']} flex="0 100px">
        <HeaderText>comment</HeaderText>
      </Box>
      <SmallText data-testid="item-comment">{data.comment}</SmallText>
    </Flex>
    <Flex pb={['s', 0]} alignItems="baseline" gridArea="browser">
      <Box display={['inline-flex', 'none']} flex="0 100px">
        <HeaderText>browser</HeaderText>
      </Box>
      <SmallText data-testid="item-browser">{`${
        data.computed_browser.Browser
      } ${data.computed_browser.Version}`}</SmallText>
    </Flex>
    <Flex pb={['s', 0]} alignItems="baseline" gridArea="device">
      <Box display={['inline-flex', 'none']} flex="0 100px">
        <HeaderText>device</HeaderText>
      </Box>
      <SmallText data-testid="item-device">{data.device}</SmallText>
    </Flex>
    <Flex pb={['s', 0]} alignItems="baseline" gridArea="platform">
      <Box display={['inline-flex', 'none']} flex="0 100px">
        <HeaderText data-testid="item-platform">platform</HeaderText>
      </Box>
      <SmallText>{data.computed_browser.Platform}</SmallText>
    </Flex>
  </StyledGrid>
))

Item.propTypes = {
  data: PropTypes.shape({
    id: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    comment: PropTypes.string.isRequired,
    computed_browser: PropTypes.shape({
      Browser: PropTypes.string.isRequired,
      Platform: PropTypes.string.isRequired,
      Version: PropTypes.string,
    }).isRequired,
    device: PropTypes.string.isRequired,
  }).isRequired,
}

export default Item
