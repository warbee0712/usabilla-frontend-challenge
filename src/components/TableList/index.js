import React from 'react'
import PropTypes from 'prop-types'
// Elements
import Flex from '../../elements/Flex'
// Components
import Header from './Header'
import TableItem from './Item'

// WARNING: THIS COMPONENT IS NOT GENERIC
// due to time constraints, this components is tightly couple
// width the given layout. To have a true generic table list
// it'd be needed more props to accept the fields, it sizes and
// and probably a render prop for the item. That is unecessary
// for this one task though.

function TableList({ items, renderItem }) {
  return (
    <Flex
      flexDirection="column"
      boxShadow="default"
      bg="white"
      data-testid="table-list"
    >
      <Header fields={['rating', 'comment', 'browser', 'device', 'platform']} />
      {items.map(renderItem)}
    </Flex>
  )
}

TableList.defaultProps = {
  // eslint-disable-next-line react/display-name
  renderItem: item => <TableItem key={item.id} data={item} />,
}

TableList.propTypes = {
  items: PropTypes.arrayOf(TableItem.propTypes.data).isRequired,
  renderItem: PropTypes.func,
}

export default TableList
