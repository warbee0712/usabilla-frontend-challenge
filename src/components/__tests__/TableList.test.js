import React from 'react'
import { render } from 'test-utils'
import TableList from '../TableList'
import Item from '../TableList/Item'
import Header from '../TableList/Header'

const ITEMS = [
  {
    comment: 'belle offre de services',
    computed_browser: {
      Browser: 'Chrome',
      Version: '32.0',
      Platform: 'MacOSX',
      FullBrowser: 'Chrome',
    },
    device: 'Desktop',
    id: '52efc552b6679cfe6ede406c',
    rating: 5,
  },
  {
    comment: 'bouton ne fonctionne pas',
    computed_browser: {
      Browser: 'Chrome',
      Version: '32.0',
      Platform: 'MacOSX',
      FullBrowser: 'Chrome',
    },
    device: 'Desktop',
    id: '52efc51fb6679c986b24c7bd',
    rating: 2,
  },
  {
    comment: 'new layout awesome',
    computed_browser: {
      Browser: 'Chrome',
      Version: '32.0',
      Platform: 'MacOSX',
      FullBrowser: 'Chrome',
    },
    device: 'Desktop',
    id: '52efc4dcfa8d1bda61ec1c64',
    rating: 4,
  },
  {
    comment: 'sugestia',
    computed_browser: {
      Browser: 'Chrome',
      Version: '32.0',
      Platform: 'Win7',
      FullBrowser: 'Chrome',
    },
    device: 'Desktop',
    id: '52ef53d2b6679c4a3cdb593c',
    rating: 3,
  },
]

describe('TableList', () => {
  it('should match snapshot', () => {
    const { container } = render(<TableList items={ITEMS} />)

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render all items', () => {
    const { getAllByTestId } = render(<TableList items={ITEMS} />)

    const items = getAllByTestId('item')

    expect(items).toHaveLength(ITEMS.length)
  })
  describe('Item', () => {
    it('should match snapshot', () => {
      const { container } = render(<Item data={ITEMS[0]} />)

      expect(container.firstChild).toMatchSnapshot()
    })
  })
  describe('Header', () => {
    it('should match snapshot', () => {
      const { container } = render(
        <Header fields={['field1', 'field2', 'field3']} />
      )

      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
