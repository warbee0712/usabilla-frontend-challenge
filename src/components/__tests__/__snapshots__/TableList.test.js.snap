// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableList Header should match snapshot 1`] = `
.c1 {
  grid-area: field1;
}

.c3 {
  grid-area: field2;
}

.c4 {
  grid-area: field3;
}

.c0 {
  display: none;
  padding-left: 1.4375rem;
  padding-right: 1.4375rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
  grid-auto-columns: 1fr 4fr 2fr 2fr 2fr;
  grid-column-gap: 1rem;
  grid-template-areas: "rating comment browser device platform";
}

.c2 {
  font-family: Milo,BlinkMacSystemFont,-apple-system,Segoe UI,Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  line-height: normal;
  text-transform: lowercase;
  font-variant: small-caps;
  font-weight: 500;
}

@media screen and (min-width:40em) {
  .c0 {
    display: grid;
  }
}

<div
  class="c0"
  display="none,grid"
>
  <div
    class="c1"
  >
    <p
      class="c2"
    >
      field1
    </p>
  </div>
  <div
    class="c3"
  >
    <p
      class="c2"
    >
      field2
    </p>
  </div>
  <div
    class="c4"
  >
    <p
      class="c2"
    >
      field3
    </p>
  </div>
</div>
`;

exports[`TableList Item should match snapshot 1`] = `
.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 0 100px;
  -ms-flex: 0 100px;
  flex: 0 100px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-area: rating;
  padding-bottom: 0.6875rem;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-area: comment;
  padding-bottom: 0.6875rem;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-area: browser;
  padding-bottom: 0.6875rem;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-area: device;
  padding-bottom: 0.6875rem;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-area: platform;
  padding-bottom: 0.6875rem;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c3 {
  font-family: Milo,BlinkMacSystemFont,-apple-system,Segoe UI,Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  line-height: normal;
  text-transform: lowercase;
  font-variant: small-caps;
  font-weight: 500;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  color: #59636b;
  border: 1px solid #59636b;
  border-radius: 9999em;
  -webkit-transition: color 0.2s cubic-bezier(0.38,1.59,0.82,0.96), box-shadow 0.2s cubic-bezier(0.38,1.59,0.82,0.96);
  transition: color 0.2s cubic-bezier(0.38,1.59,0.82,0.96), box-shadow 0.2s cubic-bezier(0.38,1.59,0.82,0.96);
  position: relative;
  font-size: 0.8125rem;
  height: 30px;
  width: 30px;
  background-color: #00a5c9;
  border-color: #00a5c9;
  color: #fff;
}

.c0 {
  display: grid;
  padding: 1.4375rem;
  grid-auto-columns: 1fr;
  grid-column-gap: 1rem;
  grid-template-areas: "rating" "comment" "browser" "device" "platform";
  border-bottom: 1px solid #eef3f7;
}

.c6 {
  font-family: BlinkMacSystemFont,-apple-system,Segoe UI,Roboto,Helvetica,Arial,sans-serif;
  font-size: 0.8125rem;
  line-height: 1.1875rem;
}

@media screen and (min-width:40em) {
  .c2 {
    display: none;
  }
}

@media screen and (min-width:40em) {
  .c1 {
    padding-bottom: 0px;
  }
}

@media screen and (min-width:40em) {
  .c5 {
    padding-bottom: 0px;
  }
}

@media screen and (min-width:40em) {
  .c7 {
    padding-bottom: 0px;
  }
}

@media screen and (min-width:40em) {
  .c8 {
    padding-bottom: 0px;
  }
}

@media screen and (min-width:40em) {
  .c9 {
    padding-bottom: 0px;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    grid-auto-columns: 1fr 4fr 2fr 2fr 2fr;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    grid-template-areas: "rating comment browser device platform";
  }
}

<div
  class="c0"
  data-testid="item"
  display="grid"
>
  <div
    class="c1"
    display="flex"
  >
    <div
      class="c2"
      display="inline-flex,none"
    >
      <p
        class="c3"
      >
        rating
      </p>
    </div>
    <span
      class="c4"
      data-testid="item-rating"
      font-size="xs"
      size="30"
    >
      5
    </span>
  </div>
  <div
    class="c5"
    display="flex"
  >
    <div
      class="c2"
      display="inline-flex,none"
    >
      <p
        class="c3"
      >
        comment
      </p>
    </div>
    <p
      class="c6"
      data-testid="item-comment"
    >
      belle offre de services
    </p>
  </div>
  <div
    class="c7"
    display="flex"
  >
    <div
      class="c2"
      display="inline-flex,none"
    >
      <p
        class="c3"
      >
        browser
      </p>
    </div>
    <p
      class="c6"
      data-testid="item-browser"
    >
      Chrome 32.0
    </p>
  </div>
  <div
    class="c8"
    display="flex"
  >
    <div
      class="c2"
      display="inline-flex,none"
    >
      <p
        class="c3"
      >
        device
      </p>
    </div>
    <p
      class="c6"
      data-testid="item-device"
    >
      Desktop
    </p>
  </div>
  <div
    class="c9"
    display="flex"
  >
    <div
      class="c2"
      display="inline-flex,none"
    >
      <p
        class="c3"
        data-testid="item-platform"
      >
        platform
      </p>
    </div>
    <p
      class="c6"
    >
      MacOSX
    </p>
  </div>
</div>
`;

exports[`TableList should match snapshot 1`] = `
.c2 {
  grid-area: rating;
}

.c4 {
  grid-area: comment;
}

.c5 {
  grid-area: browser;
}

.c6 {
  grid-area: device;
}

.c7 {
  grid-area: platform;
}

.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 0 100px;
  -ms-flex: 0 100px;
  flex: 0 100px;
}

.c0 {
  background-color: #fff;
  box-shadow: 0 1px 2px rgba(0,0,0,0.15);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-area: rating;
  padding-bottom: 0.6875rem;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-area: comment;
  padding-bottom: 0.6875rem;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-area: browser;
  padding-bottom: 0.6875rem;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-area: device;
  padding-bottom: 0.6875rem;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-area: platform;
  padding-bottom: 0.6875rem;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c1 {
  display: none;
  padding-left: 1.4375rem;
  padding-right: 1.4375rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
  grid-auto-columns: 1fr 4fr 2fr 2fr 2fr;
  grid-column-gap: 1rem;
  grid-template-areas: "rating comment browser device platform";
}

.c3 {
  font-family: Milo,BlinkMacSystemFont,-apple-system,Segoe UI,Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  line-height: normal;
  text-transform: lowercase;
  font-variant: small-caps;
  font-weight: 500;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  color: #59636b;
  border: 1px solid #59636b;
  border-radius: 9999em;
  -webkit-transition: color 0.2s cubic-bezier(0.38,1.59,0.82,0.96), box-shadow 0.2s cubic-bezier(0.38,1.59,0.82,0.96);
  transition: color 0.2s cubic-bezier(0.38,1.59,0.82,0.96), box-shadow 0.2s cubic-bezier(0.38,1.59,0.82,0.96);
  position: relative;
  font-size: 0.8125rem;
  height: 30px;
  width: 30px;
  background-color: #00a5c9;
  border-color: #00a5c9;
  color: #fff;
}

.c8 {
  display: grid;
  padding: 1.4375rem;
  grid-auto-columns: 1fr;
  grid-column-gap: 1rem;
  grid-template-areas: "rating" "comment" "browser" "device" "platform";
  border-bottom: 1px solid #eef3f7;
}

.c13 {
  font-family: BlinkMacSystemFont,-apple-system,Segoe UI,Roboto,Helvetica,Arial,sans-serif;
  font-size: 0.8125rem;
  line-height: 1.1875rem;
}

@media screen and (min-width:40em) {
  .c10 {
    display: none;
  }
}

@media screen and (min-width:40em) {
  .c9 {
    padding-bottom: 0px;
  }
}

@media screen and (min-width:40em) {
  .c12 {
    padding-bottom: 0px;
  }
}

@media screen and (min-width:40em) {
  .c14 {
    padding-bottom: 0px;
  }
}

@media screen and (min-width:40em) {
  .c15 {
    padding-bottom: 0px;
  }
}

@media screen and (min-width:40em) {
  .c16 {
    padding-bottom: 0px;
  }
}

@media screen and (min-width:40em) {
  .c1 {
    display: grid;
  }
}

@media screen and (min-width:40em) {
  .c8 {
    grid-auto-columns: 1fr 4fr 2fr 2fr 2fr;
  }
}

@media screen and (min-width:40em) {
  .c8 {
    grid-template-areas: "rating comment browser device platform";
  }
}

<div
  class="c0"
  data-testid="table-list"
  display="flex"
>
  <div
    class="c1"
    display="none,grid"
  >
    <div
      class="c2"
    >
      <p
        class="c3"
      >
        rating
      </p>
    </div>
    <div
      class="c4"
    >
      <p
        class="c3"
      >
        comment
      </p>
    </div>
    <div
      class="c5"
    >
      <p
        class="c3"
      >
        browser
      </p>
    </div>
    <div
      class="c6"
    >
      <p
        class="c3"
      >
        device
      </p>
    </div>
    <div
      class="c7"
    >
      <p
        class="c3"
      >
        platform
      </p>
    </div>
  </div>
  <div
    class="c8"
    data-testid="item"
    display="grid"
  >
    <div
      class="c9"
      display="flex"
    >
      <div
        class="c10"
        display="inline-flex,none"
      >
        <p
          class="c3"
        >
          rating
        </p>
      </div>
      <span
        class="c11"
        data-testid="item-rating"
        font-size="xs"
        size="30"
      >
        5
      </span>
    </div>
    <div
      class="c12"
      display="flex"
    >
      <div
        class="c10"
        display="inline-flex,none"
      >
        <p
          class="c3"
        >
          comment
        </p>
      </div>
      <p
        class="c13"
        data-testid="item-comment"
      >
        belle offre de services
      </p>
    </div>
    <div
      class="c14"
      display="flex"
    >
      <div
        class="c10"
        display="inline-flex,none"
      >
        <p
          class="c3"
        >
          browser
        </p>
      </div>
      <p
        class="c13"
        data-testid="item-browser"
      >
        Chrome 32.0
      </p>
    </div>
    <div
      class="c15"
      display="flex"
    >
      <div
        class="c10"
        display="inline-flex,none"
      >
        <p
          class="c3"
        >
          device
        </p>
      </div>
      <p
        class="c13"
        data-testid="item-device"
      >
        Desktop
      </p>
    </div>
    <div
      class="c16"
      display="flex"
    >
      <div
        class="c10"
        display="inline-flex,none"
      >
        <p
          class="c3"
          data-testid="item-platform"
        >
          platform
        </p>
      </div>
      <p
        class="c13"
      >
        MacOSX
      </p>
    </div>
  </div>
  <div
    class="c8"
    data-testid="item"
    display="grid"
  >
    <div
      class="c9"
      display="flex"
    >
      <div
        class="c10"
        display="inline-flex,none"
      >
        <p
          class="c3"
        >
          rating
        </p>
      </div>
      <span
        class="c11"
        data-testid="item-rating"
        font-size="xs"
        size="30"
      >
        2
      </span>
    </div>
    <div
      class="c12"
      display="flex"
    >
      <div
        class="c10"
        display="inline-flex,none"
      >
        <p
          class="c3"
        >
          comment
        </p>
      </div>
      <p
        class="c13"
        data-testid="item-comment"
      >
        bouton ne fonctionne pas
      </p>
    </div>
    <div
      class="c14"
      display="flex"
    >
      <div
        class="c10"
        display="inline-flex,none"
      >
        <p
          class="c3"
        >
          browser
        </p>
      </div>
      <p
        class="c13"
        data-testid="item-browser"
      >
        Chrome 32.0
      </p>
    </div>
    <div
      class="c15"
      display="flex"
    >
      <div
        class="c10"
        display="inline-flex,none"
      >
        <p
          class="c3"
        >
          device
        </p>
      </div>
      <p
        class="c13"
        data-testid="item-device"
      >
        Desktop
      </p>
    </div>
    <div
      class="c16"
      display="flex"
    >
      <div
        class="c10"
        display="inline-flex,none"
      >
        <p
          class="c3"
          data-testid="item-platform"
        >
          platform
        </p>
      </div>
      <p
        class="c13"
      >
        MacOSX
      </p>
    </div>
  </div>
  <div
    class="c8"
    data-testid="item"
    display="grid"
  >
    <div
      class="c9"
      display="flex"
    >
      <div
        class="c10"
        display="inline-flex,none"
      >
        <p
          class="c3"
        >
          rating
        </p>
      </div>
      <span
        class="c11"
        data-testid="item-rating"
        font-size="xs"
        size="30"
      >
        4
      </span>
    </div>
    <div
      class="c12"
      display="flex"
    >
      <div
        class="c10"
        display="inline-flex,none"
      >
        <p
          class="c3"
        >
          comment
        </p>
      </div>
      <p
        class="c13"
        data-testid="item-comment"
      >
        new layout awesome
      </p>
    </div>
    <div
      class="c14"
      display="flex"
    >
      <div
        class="c10"
        display="inline-flex,none"
      >
        <p
          class="c3"
        >
          browser
        </p>
      </div>
      <p
        class="c13"
        data-testid="item-browser"
      >
        Chrome 32.0
      </p>
    </div>
    <div
      class="c15"
      display="flex"
    >
      <div
        class="c10"
        display="inline-flex,none"
      >
        <p
          class="c3"
        >
          device
        </p>
      </div>
      <p
        class="c13"
        data-testid="item-device"
      >
        Desktop
      </p>
    </div>
    <div
      class="c16"
      display="flex"
    >
      <div
        class="c10"
        display="inline-flex,none"
      >
        <p
          class="c3"
          data-testid="item-platform"
        >
          platform
        </p>
      </div>
      <p
        class="c13"
      >
        MacOSX
      </p>
    </div>
  </div>
  <div
    class="c8"
    data-testid="item"
    display="grid"
  >
    <div
      class="c9"
      display="flex"
    >
      <div
        class="c10"
        display="inline-flex,none"
      >
        <p
          class="c3"
        >
          rating
        </p>
      </div>
      <span
        class="c11"
        data-testid="item-rating"
        font-size="xs"
        size="30"
      >
        3
      </span>
    </div>
    <div
      class="c12"
      display="flex"
    >
      <div
        class="c10"
        display="inline-flex,none"
      >
        <p
          class="c3"
        >
          comment
        </p>
      </div>
      <p
        class="c13"
        data-testid="item-comment"
      >
        sugestia
      </p>
    </div>
    <div
      class="c14"
      display="flex"
    >
      <div
        class="c10"
        display="inline-flex,none"
      >
        <p
          class="c3"
        >
          browser
        </p>
      </div>
      <p
        class="c13"
        data-testid="item-browser"
      >
        Chrome 32.0
      </p>
    </div>
    <div
      class="c15"
      display="flex"
    >
      <div
        class="c10"
        display="inline-flex,none"
      >
        <p
          class="c3"
        >
          device
        </p>
      </div>
      <p
        class="c13"
        data-testid="item-device"
      >
        Desktop
      </p>
    </div>
    <div
      class="c16"
      display="flex"
    >
      <div
        class="c10"
        display="inline-flex,none"
      >
        <p
          class="c3"
          data-testid="item-platform"
        >
          platform
        </p>
      </div>
      <p
        class="c13"
      >
        Win7
      </p>
    </div>
  </div>
</div>
`;
